### ISSUE TITLE
Changefeed - scale out support

### ISSUE THREAD

**Issue by zbynek001** [Azure/azure-cosmos-dotnet-v3#2759]
**Is your feature request related to a problem? Please describe.**
Long running `ChangeFeedIterator` has no support for scale-out scenario, e.g. when data and nr. of partitions is growing, it's not possible to split the `FeedIterator` when partition was split.

In our scenario, we create new container, iterate `FeedRange`s, and for each `FeedRange`, we create `FeedIterator`. So at the beginning we end up with very few `FeedIterator`s.
As the data grow to thousands of partitions, we still end up with the initial number of `FeedIterator`s and no way to scale-out.
And the processing is a bit CPU/IO heavy to process everything on one server.

**Describe the solution you'd like**
Best case scenario would be notified, when the partition split happens, which `FeedRange` was split and what are the new `FeedRange`s.
Or to be able to check if `FeedRange` represent multiple partitions with possibility to split it into multiple `FeedRange`s

**Describe alternatives you've considered**
Changefeed processor. Not good for our use case. We need to be more in control how things are processed. e.g. more realtime processing, some ordering/dependency logic between partitions etc. And would require duplicating some of our infrastructure.

**Additional context**

**Comment by ealsur** [comment]
@zbynek001 When looking at the processing of the change feed events, how is the time spent once the changes are received? Which is the biggest time sink in the processing?

The FeedRanges do not necessarily represent 1 partition (though they might), based on this decoupling, if there were no relationship with the number of partitions, and you could scale up to any number of iterators, how would you decide this number and why?

**Comment by zbynek001** [comment]
@ealsur we're running multiple different types of processing, each using different resources, some use only IO, some also need more cpu. At the end, we're not able to process the changefeeds without scaling-out.

Well, we don't necessarily need to do the scale-out on partition level. It was done that way, because it was the only way how to parallelize the changefeed in v2. And it has an advantage, that the changefeed of one partition represent the documents in the order they were created, which we're using for some additional logic.
Basically, as long as we're able to scale-out the processing to some desired number of parallel operation then it'd be good. 

If the FeedRange would not represent partitions, what would then `GetFeedRangesAsync` return? In such scenario might be also good to support not only splitting of FeedRange, but also merging them together, but that might lead to different discussions.

**Comment by zbynek001** [comment]
When one vm cannot handle the amount of data it needs to process from changefeed than it needs to scale out. 

Whether GetFeedRangesAsync returns FeedRanges representing partitions or not is not really important. What's important is to be able to split individual FeedRanges independently from other and to be able to continue processing where we left off before the split. One such scenario is when adding more instances to vm scaleset and wanting to distribute the load also to the new instances without as small interuption of processing as possible, i.e. without recreating all of the FeedRangeIterators
